pytorch_lightning/utilities/distributed.py

```diff
@@ -376,3 +376,30 @@ def init_dist_connection(
             f"All distributed processes registered. Starting with {world_size} processes\n"
             f"{'-' * 100}\n"
         )
+
+
+def _collect_states_on_rank_zero(state: Dict[str, Any], device: torch.device) -> Optional[Dict[int, Any]]:
```

Comment:
brainstorm question: should we have this be part of the the Accelerator(Device)? gpu vs tpu vs ipu will have different collective

Reply:
Yes, but this is still experimental and we don't need to add support for all plugins yet.